LinkList.java

```diff
@@ -120,34 +133,153 @@ public boolean contains(Object o) {
 
      }
 
-     public static void main(String [ ] args) {
-        ArrayList a=new LinkList();
+      public Iterator<E> iterator() {
+        return null;
+    }  
+
+    public void addFirst(E e) {
+        if(size()==0){
+            head.data=e;
+            numNodes++;
+            return;
+        }
+        Node holder=new Node<E>();
+        holder.data=head.data;
+        holder.next=head.next;
+        head.data=e;
+        head.next=holder;
+        numNodes++;
+    }
+
+    public void addLast(E e) {
+        if(size()==0){
+            head.data=e;
+            numNodes++;
+            return;
+        }
+        Node tra=head;
+        while(tra.next!=null) {
+            tra=tra.next;
+        }
+        Node temp=new Node<E>();
+        temp.data=e;
+        tra.next=temp;
+        numNodes++;
+    }
+
+    @SuppressWarnings("unchecked")
+    public E getFirst() {
```

Comment:
Is it supposed to return null or throw exception? Please double check the doc.

Reply:
throw new NoSuchElementException(); Fixed